test/integration/spec/localtrackpublication.js

```diff
@@ -121,6 +124,12 @@ describe('LocalTrackPublication', function() {
             return tracksRemoved(thatParticipant, thisParticipant.tracks.size);
           }));
 
+          // NOTE(mmalavalli): Even though the "trackUnpublished" events are
+          // fired on the RemoteParticipants, we need to make sure that the
+          // SDP negotiation is complete before we re-publish the LocalTrack.
+          // Therefore we wait for 2 seconds.
+          await waitFor(5000);
```

Comment:
Looks like thisis  something that app developers need to be aware off as well - could we fix this in sdk by delaying the trackUnpublished event for SDP negotiation completion?